lib/Models/Catalog/CatalogItems/CartoMapV3CatalogItem.ts

```diff
@@ -0,0 +1,290 @@
+    return new CartoMapV3Stratum(newModel as CartoMapV3CatalogItem) as this;
+  }
+
+  // Hide "cartodb_id" style
+  @computed get styles() {
+    return [
+      createStratumInstance(TableStyleTraits, {
+        id: "cartodb_id",
+        hidden: true
+      })
+    ];
+  }
+}
+
+StratumOrder.addLoadStratum(CartoMapV3Stratum.stratumName);
+
+export default class CartoMapV3CatalogItem extends GeoJsonMixin(
+  CreateModel(CartoMapV3CatalogItemTraits)
+) {
+  static readonly type = "carto-v3";
+
+  @observable
+  geoJsonUrls: string[] = [];
+
+  @observable
+  mvtTileJsonUrls: string[] = [];
+
+  @observable geoJsonSize: number | undefined;
+
+  constructor(
+    id: string | undefined,
+    terria: Terria,
+    sourceReference?: BaseModel | undefined
+  ) {
+    super(id, terria, sourceReference);
+
+    if (this.strata.get(CartoMapV3Stratum.stratumName) === undefined) {
+      runInAction(() => {
+        this.strata.set(
+          CartoMapV3Stratum.stratumName,
+          new CartoMapV3Stratum(this)
+        );
+      });
+    }
+  }
+
+  get type() {
+    return CartoMapV3CatalogItem.type;
+  }
+
+  get typeName() {
+    return i18next.t("models.carto-v3.name");
+  }
+
+  protected async forceLoadMetadata() {
+    let response: JsonObject | undefined;
+
+    /** If cartoQuery is defined - use Query API (https://api-docs.carto.com/#8f2020d9-edf3-4b50-ae58-9edeaa34613c)
+     */
+    if (this.cartoQuery) {
+      const url = new URI(this.baseUrl)
+        .path("")
+        .path(`v3/maps/${this.connectionName}/query`);
+
+      response = (
+        await callCartoApi(url.toString(), this.accessToken, {
+          q: this.cartoQuery,
+          geo_column: this.cartoGeoColumn
+        })
+      ).throwIfError();
+    }
+    // If cartoTableName is defined - use Table API (https://api-docs.carto.com/#6a05d4d7-c6a1-4635-a8de-c91fa5e77fda)
+    else if (this.cartoTableName) {
+      const url = new URI(this.baseUrl)
+        .path("")
+        .path(`v3/maps/${this.connectionName}/table`)
+        .query({
+          name: this.cartoTableName,
+          columns: this.cartoColumns?.join(","),
+          geo_column: this.cartoGeoColumn
+        });
+
+      response = (
+        await callCartoApi(url.toString(), this.accessToken)
+      ).throwIfError();
+    } else {
+      throw new TerriaError({
+        title: "Invalid Carto V3 config",
+        message: "`cartoQuery` or `cartoTableName` must be defined"
+      });
+    }
+
+    let geoJsonUrls: string[] = [];
+    let mvtTileJsonUrls: string[] = [];
+
+    if (
+      response &&
+      isJsonObject(response.geojson) &&
+      isJsonStringArray(response.geojson.url)
+    ) {
+      geoJsonUrls = response.geojson.url;
+    }
+
+    if (
+      response &&
+      isJsonObject(response.tilejson) &&
+      isJsonStringArray(response.tilejson.url)
+    ) {
+      mvtTileJsonUrls = response.tilejson.url;
+    }
+
+    if (geoJsonUrls.length === 0 /*&& mvtTileJsonUrls.length === 0*/) {
```

Comment:
Can we remove commented out bit now?

Reply:
We need to leave it as a comment until we add tile JSON support